src/aspire/basis/polar_2d.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import numpy as np
+import finufftpy
+
+from aspire.utils import ensure
+from aspire.utils.matrix import roll_dim, unroll_dim
+from aspire.utils.matlab_compat import m_reshape
+from aspire.basis import Basis
+
+logger = logging.getLogger(__name__)
+
+
+class PolarBasis2D(Basis):
+    """
+    Define a derived class for polar Fourier representation for 2D images
+    """
+
+    def __init__(self, size, nrad=None, ntheta=None):
+        """
+        Initialize an object for the 2D polar Fourier grid class
+
+        :param size: The shape of the vectors for which to define the grid.
+            Currently only square images are supported.
+        :param nrad: The number of points in the radial dimension.
+        :param ntheta: The number of points in the angular dimension.
+        """
+
+        ndim = len(size)
+        ensure(ndim == 2, 'Only two-dimensional grids are supported.')
+        ensure(len(set(size)) == 1, 'Only square domains are supported.')
+
+        self.nrad = nrad
+        if nrad is None:
+            self.nrad = self.nres // 2
+
+        self.ntheta = ntheta
+        if ntheta is None:
+            # try to use the same number as Fast FB basis
+            self.ntheta = 8 * self.nrad
+
+        super().__init__(size)
```

Comment:
Again, call this first.

Reply:
Same as other basis classes, the base constructor is designed to be called at the end of the derived constructor because the `_build` function is included in the base constructor.